src/cpu/instr.rs

```diff
@@ -78,3 +332,29 @@ pub fn execute(enc: &Encoding, cpu: &mut CPU) {
     
     (op.execute)(enc, cpu);
 }
+
+fn gen_flag(is_subtract: bool, var_1: u32, var_2: u32, mut var_out: u32) -> u16 {
+    let mut temp_flag = Flags::CLEAR_FLAG;
+    if extract(var_out, 15, 1) == 1 { temp_flag |= Flags::N; }
+    
+    if extract(var_out, 16, 1) == 1 { temp_flag |= Flags::C; }
+
+    if is_subtract {
+        if (extract(var_1, 15, 1) ^ extract(var_2, 15, 1) ^ 1) & (extract(var_1, 15, 1) ^ extract(var_out, 15, 1)) == 1 { temp_flag |= Flags::O; }
+    }
+    else {
+        if (extract(var_1, 15, 1) ^ extract(var_2, 15, 1) ^ 0) & (extract(var_1, 15, 1) ^ extract(var_out, 15, 1)) == 1 { temp_flag |= Flags::O; }
+    }
```

Comment:
Avoid code duplication, it can be simpified to something like `(extract(var_1, 15, 1) ^ extract(var_2, 15, 1) ^ (is_substract as u32)) & (extract(var_1, 15, 1) ^ extract(var_out, 15, 1))`